### PR TITLE
feat[smock]: add support for receive

### LINKED ISSUE #1245: Smockit - support mocking receive() and low-level calls
**Is your feature request related to a problem? Please describe.**
`smockit` allows stubbing/mocking responses for unit testing. It currently supports mocking defined functions, state variables and `fallback` of a given contract, but doesn't seem to support Solidity 0.8.0 [`receive`](https://docs.soliditylang.org/en/v0.8.0/contracts.html#receive-ether-function). 

For example: a `contract` with: 

```solidity
//Bank.sol
contract Bank {
    uint balance;
    function withdraw(){
         msg.sender.transfer(balance);
    }
    receive() external payable {
         balance += msg.value;
    } 
}
```

mocked as :
```javascript
//Bank.test.js
    this.BankFactory = await ethers.getContractFactory('Bank');
    this.bank = await this.BankFactory.deploy();
    await this.bank.deployed();
    const mockBank = await smockit(this.bank);
```
will return an object `mockBank` with: 

```javascript
//console.log(mockBank);

Contract {
    ...,
    smocked: {
        fallback:  {...},
        'withdraw()': {...},
        balance: {...}

    }
}
```
As a result, it is not possible to mock the `receive()` function of the bank. 

`mockBank.smocked.receive.will.return.with((true))` returns an error message `TypeError: Cannot read property 'will' of undefined`. 

**Describe the solution you'd like**
Support mocking the `receive()` function to return the standard response `(bool, bytes memory)`. 

**Describe alternatives you've considered**
I've looked through the docs but didn't see a way to mock `receive`, `call` etc. 

**Additional context**

### PR DESCRIPTION
**Pull request by smartcontracts**
**Description**
Adds naive support for the Solidity receive function as an alias to fallback. Actual Solidity behavior is a bit different but ethers throws out fallback/receive from the ABI when it creates the contract object which limits how much we can do. I think this is good enough for now.

**Metadata**
- Fixes #1245

### CODE REVIEW

File: packages/smock/test/smockit/function-manipulation.spec.ts [hunk 1]
```diff
@@ -115,6 +115,107 @@ describe('[smock]: function manipulation tests', () => {
...
+      mock.smocked.receive.will.return()
+
+      expect(
+        await ethers.provider.call({
+          to: mock.address,
+        })
+      ).to.equal(expected)
+    })
+
+    describe.skip('resetting the receive function', () => {
```
Comment (snario) [inline packages/smock/test/smockit/function-manipulation.spec.ts:197]: Why is this skipped?

Comment (smartcontracts) [reply packages/smock/test/smockit/function-manipulation.spec.ts:197]: `reset` is a function I plan to add eventually(tm) but haven't done yet. So just adding the test as a reminder to myself.

File: packages/smock/test/smockit/function-manipulation.spec.ts [hunk 2]
```diff
@@ -115,6 +115,107 @@ describe('[smock]: function manipulation tests', () => {
...
+      mock.smocked.receive.will.revert.with(expected)
+
+      await expect(
+        ethers.provider.call({
+          to: mock.address,
+        })
+      ).to.be.revertedWith(expected)
+    })
+
+    it('should be able to make a receive function emit an event', async () => {
```
Comment (tynes) [inline packages/smock/test/smockit/function-manipulation.spec.ts:174]: This should be implemented or skipped

File: packages/smock/test/smockit/function-manipulation.spec.ts (hunk 1 above)
Comment (tynes) [reply packages/smock/test/smockit/function-manipulation.spec.ts:197]: you should add a comment here saying that

**Review by tynes — APPROVED** [review]

**Comment by tynes** [comment]
Any updates on this @smartcontracts re: smock living in a different organization?

**Comment by smartcontracts** [comment]
> Any updates on this @smartcontracts re: smock living in a different organization?

Yeah smock is going to live at defi-wonderland/smock. Going to close this in favor of using the other repo.

**Pull request closed without merging**